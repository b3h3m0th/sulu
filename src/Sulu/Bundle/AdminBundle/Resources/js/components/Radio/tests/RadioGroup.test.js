// @flow
import {fireEvent, render, screen} from '@testing-library/react';
import React from 'react';
import RadioGroup from '../RadioGroup';
import Radio from '../Radio';

test('The component should render', () => {
    const {container} = render(
        <RadioGroup className="my-group" value="1">
            <Radio value="1" />
            <Radio value="2" />
            <Radio value="3" />
        </RadioGroup>
    );
    expect(container).toMatchSnapshot();
});

test('The component should check the correct radio', () => {
    render(
        <RadioGroup value="1">
            <Radio value="1" />
            <Radio value="2" />
            <Radio value="3" />
        </RadioGroup>
    );

    const radioGroup = [
        screen.queryByDisplayValue('1'),
        screen.queryByDisplayValue('2'),
        screen.queryByDisplayValue('3'),
    ];

    expect(radioGroup[0]).toBeChecked();
    expect(radioGroup[1]).not.toBeChecked();
    expect(radioGroup[2]).not.toBeChecked();
});

test('The component should pass the disabled state to the radios', () => {
    render(
        <RadioGroup disabled={true} value="1">
            <Radio value="1" />
            <Radio value="2" />
            <Radio value="3" />
        </RadioGroup>
    );

    const radioGroup = [
        screen.queryByDisplayValue('1'),
        screen.queryByDisplayValue('2'),
        screen.queryByDisplayValue('3'),
    ];

    expect(radioGroup[0]).toBeDisabled();
    expect(radioGroup[1]).toBeDisabled();
    expect(radioGroup[2]).toBeDisabled();
});

test('The component should pass the change callback to the radios', () => {
    const onChange = jest.fn();
    render(
        <RadioGroup onChange={onChange}>
            <Radio value="1" />
            <Radio value="2" />
            <Radio value="3" />
        </RadioGroup>
    );

    const radioGroup = [
        screen.queryByDisplayValue('1'),
        screen.queryByDisplayValue('2'),
        screen.queryByDisplayValue('3'),
    ];

    fireEvent.click(radioGroup[0]);
    fireEvent.click(radioGroup[1]);
    fireEvent.click(radioGroup[2]);

    expect(onChange).toHaveBeenCalledTimes(3);
});
