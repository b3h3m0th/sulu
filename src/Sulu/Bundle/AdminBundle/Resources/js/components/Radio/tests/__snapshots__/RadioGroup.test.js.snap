// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render 1`] = `
<div>
  <div
    class="my-group"
  >
    <label
      class="label"
      tabindex="-1"
    >
      <span
        class="switch radio dark"
      >
        <input
          checked=""
          type="radio"
          value="1"
        />
        <span />
      </span>
    </label>
    <label
      class="label"
      tabindex="-1"
    >
      <span
        class="switch radio dark"
      >
        <input
          type="radio"
          value="2"
        />
        <span />
      </span>
    </label>
    <label
      class="label"
      tabindex="-1"
    >
      <span
        class="switch radio dark"
      >
        <input
          type="radio"
          value="3"
        />
        <span />
      </span>
    </label>
  </div>
</div>
`;
